### ISSUE TITLE
improve (matplotlib.widgets.Slider) doc + input validation

### ISSUE THREAD

**Issue by sbowa** [matplotlib/matplotlib#7687]
This minimal example throws an error (without crashing) when
1) the slider is initialized with a slidermin/slidermax value
2) the slider is moved with the mouse
(the program does not crash, but the slider also does not respond)

Minimal example:
```python
import matplotlib.pyplot as plt
from matplotlib.widgets import Slider

fig, ax = plt.subplots()
slider = Slider(ax=ax, label='', valmin=0.0, valmax=24.0, valinit=15.0,
                slidermin=10.0, slidermax=20.0)
plt.show()
```

Fix follows.

**Comment by tacaswell** [comment]
@sbowa Can you open a pull request with that change?

**Comment by tacaswell** [comment]
This is not a bug, but a miss-use of the library.  The `slidermin` and `slidermax` are either `None` or another slider instance (see http://matplotlib.org/api/widgets_api.html#matplotlib.widgets.Slider) not a float.

**Comment by tacaswell** [comment]
The exact work to do on this:

 - update `Slider.__init__` docstring to numpydoc (so it renders better)
 - add a duck-type validation of slidermin/slidermax (something like `if sm is not None or hasattr(sm, 'val'): raise`

**Comment by tacaswell** [comment]
@sbowa no problem!  Sorry for not noticing what was going on sooner.

**Comment by tacaswell** [comment]
Closed by #8134
